ArmPkg/Drivers/MmCommunicationDxe/MmCommunication.c

```diff
@@ -36,6 +43,99 @@ STATIC EFI_EVENT  mSetVirtualAddressMapEvent;
 //
 STATIC EFI_HANDLE  mMmCommunicateHandle;
 
+/**
+  Send mm communicate request via FF-A.
+
+  @retval EFI_SUCCESS
+  @retval Others                   Error.
+
+**/
+STATIC
+EFI_STATUS
+EFIAPI
+SendFfaMmCommunicate (
+  IN VOID
+  )
+{
+  EFI_STATUS       Status;
+  DIRECT_MSG_ARGS  CommunicateArgs;
+
+  ZeroMem (&CommunicateArgs, sizeof (DIRECT_MSG_ARGS));
+
+  CommunicateArgs.Arg0 = (UINTN)mNsCommBuffMemRegion.PhysicalBase;
+
+  Status = ArmFfaLibMsgSendDirectReq (
+             mStMmPartId,
+             0,
+             &CommunicateArgs
+             );
+
+  while (Status == EFI_INTERRUPT_PENDING) {
+    // We are assuming vCPU0 of the StMM SP since it is UP.
+    Status = ArmFfaLibRun (mPartId, 0x00);

```

Comment:
Sorry for the late reply. I found a few new issues with the updated hafnium. You can ignore them at this moment. But with the fix will be good to have.

This should have been:
```suggestion
    Status = ArmFfaLibRun (mStMmPartId, 0x00);
```

Reply:
Thanks to let me know :) And Sorry for this mistake..!